[api] Compiler format mismatch warnings

### The problem

With enabled strict compiler warnings as build flags (`-Wall -Wextra -Werror`):
`api` component to compile due to two format mismatches.

Immediately obvious by trying out simple config YAML below.

Likely they have no fatal impact. However, code base best to compile without such warnings if intended to be used locally for compiling whole tree. And most other components apart from `api` compile without issues (maybe followup ticket for other observation).

### Which version of ESPHome has the issue?

2026.1.5

### What type of installation are you using?

pip

### What platform are you using?

ESP32

### Component causing the issue

api

### YAML Config

```yaml
esphome:
  name: compileerror
  platformio_options:
    build_flags:
      - -Wall -Wextra -Werror

esp32:
  board: esp32-c6-devkitc-1
  variant: esp32c6

network:

api:

logger:
  level: VERBOSE
```

### Anything in the logs that might be useful for us?

```txt
In file included from src/esphome/core/component.h:9,
                 from src/esphome/components/api/proto.h:3,
                 from src/esphome/components/api/proto.cpp:1:
src/esphome/components/api/proto.cpp: In member function 'virtual void esphome::api::ProtoDecodableMessage::decode(const uint8_t*, size_t)':
src/esphome/components/api/proto.cpp:136:23: error: format '%u' expects argument of type 'unsigned int', but argument 5 has type 'uint32_t' {aka 'long unsigned int'} [-Werror=format=]
  136 |         ESP_LOGV(TAG, "Invalid field type %u at offset %ld", field_type, (long) (ptr - buffer));
      |                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

```txt
src/esphome/components/api/api_connection.cpp:1547:17: error: format '%lu' expects argument of type 'long unsigned int', but argument 8 has type 'int' [-Werror=format=]
 1547 |   ESP_LOGV(TAG, "Hello from client: '%s' | %s | API Version %" PRIu32 ".%" PRIu32, this->helper_->get_client_name(),
```

From code obvious that wrong specifiers were used.

### Additional information

_No response_

--- a/esphome/components/api/api_connection.cpp
+++ b/esphome/components/api/api_connection.cpp
@@ -1510,7 +1510,7 @@ bool APIConnection::send_hello_response_(const HelloRequest &msg) {
   this->client_api_version_major_ = msg.api_version_major;
   this->client_api_version_minor_ = msg.api_version_minor;
   char peername[socket::SOCKADDR_STR_LEN];
-  ESP_LOGV(TAG, "Hello from client: '%s' | %s | API Version %" PRIu32 ".%" PRIu32, this->helper_->get_client_name(),
+  ESP_LOGV(TAG, "Hello from client: '%s' | %s | API Version %" PRIu16 ".%" PRIu16, this->helper_->get_client_name(),
            this->helper_->get_peername_to(peername), this->client_api_version_major_, this->client_api_version_minor_);
 
   HelloResponse resp;

--- a/esphome/components/api/proto.cpp
+++ b/esphome/components/api/proto.cpp
@@ -133,7 +133,7 @@ void ProtoDecodableMessage::decode(const uint8_t *buffer, size_t length) {
         break;
       }
       default:
-        ESP_LOGV(TAG, "Invalid field type %u at offset %ld", field_type, (long) (ptr - buffer));
+        ESP_LOGV(TAG, "Invalid field type %" PRIu32 " at offset %ld", field_type, (long) (ptr - buffer));
         return;
     }
   }

